bugbug/models/rcatype.py

```diff
@@ -0,0 +1,181 @@
+            bug_features.landings(),
+            bug_features.blocked_bugs_number(),
+            bug_features.ever_affected(),
+            bug_features.affected_then_unaffected(),
+            bug_features.product(),
+            bug_features.component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.url(),
+            feature_cleanup.fileref(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
+                ),
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = OneVsRestClassifier(xgboost.XGBClassifier(n_jobs=16))
+
+    # return rca from a whiteboard string
+    def get_rca_from_whiteboard(self, whiteboard_data):
+        rca = []
+        whiteboard_data = re.sub(" ", "", whiteboard_data).lower()
+        for whiteboard in whiteboard_data.split("["):
+            if not any(whiteboard.startswith(s) for s in ("rca-", "rca:")):
+                continue
+
+            rca_whiteboard = re.sub("]", "", whiteboard)
+
+            # Hybrid cases: rca:X-Y
+            rca_whiteboard = re.sub(":", "-", rca_whiteboard)
+
+            rca_whiteboard_split = (
+                rca_whiteboard.split("-", 1)
+                if self.rca_subcategories_enabled
+                else rca_whiteboard.split("-")
+            )
+
+            if rca_whiteboard_split[1] not in self.RCA_LIST:
+                logger.warning(rca_whiteboard_split[1] + " not in RCA_LIST")
+            else:
+                rca.append(rca_whiteboard_split[1])
+        return rca
+
+    def get_labels(self):
+        all_rca = []
+        bug_ids = []
+        for bug in bugzilla.get_bugs():
+            bug_ids.append(int(bug["id"]))
+            all_rca.append(self.get_rca_from_whiteboard(bug["whiteboard"]))
+        classes = {}
+        for bug_id, bug_rca in zip(bug_ids, all_rca):
+            target = np.zeros(len(self.RCA_LIST))
+            for rca in bug_rca:
+                target[self.RCA_LIST.index(rca)] = 1
+            classes[bug_id] = target
```

Comment:
This code is still a bit overcomplex, why do you need to do two passes?

Something like this should work as well:
```Python
for bug in bugzilla.get_bugs():
    target = np.zeros(len(self.RCA_LIST))
    for rca in self.get_rca_from_whiteboard(bug["whiteboard"]):
        target[self.RCA_LIST.index(rca)] = 1
    classes[bug["id"]] = target
```